Global Loading

JacobWestfield opened the pull request:
Создал папку HOC, в ней компонент HOC. Перенес логику загрузки пользователей в HOC, поменял отображение App в index.js

JacobWestfield committed:
Global Loading

Создал папку HOC, в ней компонент HOC. Перенес логику загрузки пользователей в HOC, поменял отображение App в index.js

Legrank commented on src/HOC/withUsers.jsx:
```diff
@@ -0,0 +1,21 @@
+    useEffect(() => {
+        dispatch(loadUsersList());
+    }, []);
+    const users = useSelector(getUsersSelector());
+    if (users.length === 0) {
+        return <h1>Loading...</h1>;
+    }
```
У нас есть параметр isLoading, логичнее проверять не длину массива, а статус загрузки. 
Возможна ситуация с ошибкой, когда загрузка закончена а пользователей нету, но её нужно обрабатывать отдельно. Если есть желание можешь сделать проверку на ошибки, но это не обязательно.

JacobWestfield committed:
Global loading + errors

Добавил обработку ошибок через react-toastify. изменил обработку загрузки в HOC
